pkg/actuators/client/client.go

```diff
@@ -20,86 +20,419 @@ import (
+	// IC Virtual Private Cloud (VPC) API
+	vpcService, err := vpcv1.NewVpcV1(&vpcv1.VpcV1Options{
+		Authenticator: &core.IamAuthenticator{
+			ApiKey: credentialVal,
+		},
+	})
+	if err != nil {
+		return nil, err
 	}
 
-	vpcService, vpcServiceErr := vpcv1.NewVpcV1(options)
+	// IC Resource Manager API
+	resourceManagerService, err := resourcemanagerv2.NewResourceManagerV2(&resourcemanagerv2.ResourceManagerV2Options{
+		Authenticator: &core.IamAuthenticator{
+			ApiKey: credentialVal,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
 
-	if vpcServiceErr != nil {
-		panic(vpcServiceErr)
+	// IC Resource Controller API
+	resourceControllerService, err := resourcecontrollerv2.NewResourceControllerV2(&resourcecontrollerv2.ResourceControllerV2Options{
+		Authenticator: &core.IamAuthenticator{
+			ApiKey: credentialVal,
+		},
+	})
+	if err != nil {
+		return nil, err
 	}
+
 	return &ibmCloudClient{
-		vpcService: vpcService,
+		vpcService:                vpcService,
+		resourceControllerService: resourceControllerService,
+		resourceManagerService:    resourceManagerService,
 	}, nil
 }
 
-// InstanceGet returns retrieves a single instance specified by instanceID
-func (c *ibmCloudClient) InstanceGet(instanceID string) (*vpcv1.Instance, error) {
+// InstanceExistsByName checks if the instance exist in VPC
+func (c *ibmCloudClient) InstanceExistsByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) (bool, error) {
+	// Get Instance info
+	_, err := c.InstanceGetByName(name, machineProviderConfig)
+
+	// Instance found
+	if err == nil {
+		return true, nil
+	}
+
+	// Instance not found
+	if err.Error() == "Instance not found" {
+		return false, nil
+	}
+
+	// Could not retrieve Instances list
+	return false, err
+}
+
+// InstanceDeleteByName deletes the requested instance
+func (c *ibmCloudClient) InstanceDeleteByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) error {
+	// Get Instance info
+	getInstance, err := c.InstanceGetByName(name, machineProviderConfig)
+	if err != nil {
+		return err
+	}
+
+	// Get instance ID
+	instanceID := *getInstance.ID
+	if instanceID == "" {
+		return fmt.Errorf("Could not get the Instance ID")
+	}
+
+	// Initialize New Delete Instance Options
+	deleteInstanceOption := c.vpcService.NewDeleteInstanceOptions(instanceID)
+	// // Set Instance ID
+	// deleteInstanceOption.SetID(instanceID)
+
+	// Delete the Instance
+	_, err = c.vpcService.DeleteInstance(deleteInstanceOption)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// InstanceGetByName retrieves a single instance specified by Instance Name
+func (c *ibmCloudClient) InstanceGetByName(name string, machineProviderConfig *ibmcloudproviderv1.IBMCloudMachineProviderSpec) (*vpcv1.Instance, error) {
+	// Get region info
+	regionName := machineProviderConfig.Region
+	region, _, err := c.vpcService.GetRegion(c.vpcService.NewGetRegionOptions(regionName))
+	if err != nil {
```

Comment:
In addition to `err != nil`, we may want to check the response codes from the service as well.
If for some reason the region wasn't found and the service returned a 404, I'm not certain whether err would be nil or not. If err != nil and the return code was 200, we know we're good.

Reply:
Good point. I just tried it locally.

In the case of a 404, we should get a `! nil` err with a `region not found`. So we should be good.